VirtualReality/Widgets/qMRMLVirtualRealityHomeWidget.cxx

```diff
@@ -130,19 +135,157 @@ void qMRMLVirtualRealityHomeWidget::setVirtualRealityViewNode(vtkMRMLVirtualReal
 void qMRMLVirtualRealityHomeWidget::updateWidgetFromMRML()
 {
   Q_D(qMRMLVirtualRealityHomeWidget);
+  vtkMRMLVirtualRealityViewNode* vrViewNode = d->VirtualRealityViewNode;
+
+  bool wasBlocked = d->MotionSensitivitySliderWidget->blockSignals(true);
+  d->MotionSensitivitySliderWidget->setValue(vrViewNode != nullptr ? vrViewNode->GetMotionSensitivity() * 100.0 : 0);
+  d->MotionSensitivitySliderWidget->setEnabled(vrViewNode != nullptr);
+  d->MotionSensitivitySliderWidget->blockSignals(wasBlocked);
+
+  wasBlocked = d->FlySpeedSliderWidget->blockSignals(true);
+  d->FlySpeedSliderWidget->setValue(vrViewNode != nullptr ? vrViewNode->GetMotionSpeed() : 1.6666);
+  d->FlySpeedSliderWidget->setEnabled(vrViewNode != nullptr);
+  d->FlySpeedSliderWidget->blockSignals(wasBlocked);
+
+  /*
+  bool wasBlocked = d->LockMagnificationCheckBox->blockSignals(true);
+  d->LockMagnificationCheckBox->setChecked(vrViewNode != nullptr && vrViewNode->);
+  d->LockMagnificationCheckBox->setEnabled(vrViewNode != nullptr);
+  d->LockMagnificationCheckBox->blockSignals(wasBlocked);
+  */
+  //Magnification lock of view node not implemented yet 
+
+  d->Magnification001xButton->setEnabled(vrViewNode != nullptr && vrViewNode->GetMagnification() != NULL);
+  d->Magnification01xButton->setEnabled(vrViewNode != nullptr && vrViewNode->GetMagnification() != NULL);
+  d->Magnification1xButton->setEnabled(vrViewNode != nullptr && vrViewNode->GetMagnification() != NULL);
+  d->Magnification10xButton->setEnabled(vrViewNode != nullptr && vrViewNode->GetMagnification() != NULL);
+  d->Magnification100xButton->setEnabled(vrViewNode != nullptr && vrViewNode->GetMagnification() != NULL);
+}
+
+//-----------------------------------------------------------------------------
+void qMRMLVirtualRealityHomeWidget::onMotionSensitivityChanged(double percent)
+{
+  Q_D(qMRMLVirtualRealityHomeWidget);
+  vtkMRMLVirtualRealityViewNode* vrViewNode = d->VirtualRealityViewNode;
+
+  if (!vrViewNode)
+  {
+    qCritical() << Q_FUNC_INFO << " Failed: view node is null";
```

Comment:
You also need to return here obviously, otherwise the next line results in an immediate crash.